package.json

```diff
@@ -23,14 +23,16 @@
     "typescript": "^3.3.1"
   },
   "scripts": {
-    "postinstall": "lerna bootstrap && npm run -s update-greenkeeper-json",
+    "postinstall": "npm run -s update-package-lock && npm run -s update-greenkeeper-json",
```

Comment:
Before this change, `npm install` triggered `lerna bootstrap` which did incremental update of dependencies - this is usually reasonably fast.

After this change, `npm install` triggers `update-package-lock` which cleans all dependencies first and then reinstalls them from scratch. I am concerned about the impact of this change on how long it will take to run `npm install` now.

The typical scenario:
 - Fetch new changes from GitHub: `git pull`
 - Update deps, rebuild and run tests to verify all is good: `npm it`

Can we look for a different solution for updating package locks please, one that will not degrade experience in most common use cases (on the happy path)?

I also think we should open an issue in Lerna to discuss why `lerna bootstrap` is not updating package-lock files by default, and work with the Lerna team to fix it.

To get this pull request landed sooner, I am proposing to revert changes in this line.

Reply:
We'll have to pay the price to ensure `package-lock.json` files are up to date so that we benefit of less time consuming on `npm ci` for both build and local development.

There are a few situations:

1. Add or remove dependencies, or update out-of-range versions in `package.json`. We must update `package-lock.json` files.

2. There are new versions of dependencies in range of what's in `package.json` files. It's nice to have to update `package-lock.json` files.

If we don't run `npm run update-package-locks` during `npm install`. We either have to do so manually and trigger it during the release process.

Reply:
I am arguing that the case when no package-lock updates are needed is much more frequent than the case when some of package-lock files need to be updated.

Anyhow, as long as `npm cit` works fast, then I can live with slow `npm install` because I am not going to use it myself.